src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -0,0 +1,131 @@
+        .thenReturn(mockRequestDispatcher);
+
+    mockMessageStore = Mockito.mock(MessageStore.class);
+    profileServlet.setMessageStore(mockMessageStore);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    profileServlet.setUserStore(mockUserStore);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    User fakeUser = new TestUserBuilder().withName("test_user").build();
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/users/test_user");
+    Mockito.when(mockUserStore.getUser("test_user"))
+        .thenReturn(fakeUser);
+
+    String fakeUsername = "test_user";
+    Mockito.when(mockRequest.getAttribute("username")).thenReturn("test_user");
+
+    List<Message> fakeMessagesByUser = new ArrayList<>();
+    fakeMessagesByUser.add(new TestMessageBuilder().withAuthorId(fakeUser.getId()).build());
+    Mockito.when(mockMessageStore.getMessagesByUser(fakeUser.getId()))
+        .thenReturn(fakeMessagesByUser);
+
+    profileServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("messagesByUser", fakeMessagesByUser);
+    Mockito.verify(mockRequest).setAttribute("username", fakeUsername);
+    Mockito.verify(mockRequest).setAttribute("user", fakeUser);
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_UserNotLoggedIn() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn(null);
+
+    profileServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockMessageStore, Mockito.never()).addMessage(Mockito.any(Message.class));
+    Mockito.verify(mockResponse).sendRedirect("/login");
+  }
+
+  @Test
+  public void testDoPost_InvalidUser() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_user");
+    Mockito.when(mockUserStore.getUser("test_user")).thenReturn(null);
+
+    profileServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockMessageStore, Mockito.never()).addMessage(Mockito.any(Message.class));
+    Mockito.verify(mockResponse).sendRedirect("/login");
+  }
+
+  @Test
+  public void testDoPost_CleansHtmlContent() throws IOException, ServletException {
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+    Mockito.when(mockRequest.getParameter("About Me"))
+        .thenReturn("Contains <b>html</b> and <script>JavaScript</script> content.");
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/users/test_user");
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_user");
+
+    User fakeUser = new TestUserBuilder().withName("test_user").build();
+    Mockito.when(mockUserStore.getUser("test_user")).thenReturn(fakeUser);
+
+    profileServlet.doPost(mockRequest, mockResponse);
+
+    assert (mockUserStore.getUser("test_user").getAboutMe().equals("Contains html and  content."));
```

Comment:
Use `Assert.assertEquals`, which will use the fancy JUnit stuff.  If you just use Java's `assert`, then your test will crash, but not have the JUnit information.  And when you use `assertEquals`, the first argument will be the "expected" string, and the second argument will be the actual string that gets returned by calling the function.